### PR TITLE
Add "code commune" filter

### LINKED ISSUE #82: Ajouter filtre de "code commune (Insee)"

### PR DESCRIPTION
**Pull request by HAEKADI**
closes #82 
I tried using a regex to valide `code_commune` (example: `^(0[1-9]|[1-9][ABab\d])\d{3}$`) but couldn't find a general rule that takes into consideration all exceptions. Instead I used a list of all valid codes (~38k), from official Insee [dataset](https://www.insee.fr/fr/information/5057840) to avoid any errors.

### CODE REVIEW

File: aio/aio-proxy/aio_proxy/parsers/code_commune.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        code_commune_clean if valid.
+
+    Raises:
+        ValueError: if code_commune_clean not valid.
+    """
+    if code_commune_clean is None:
+        return None
+    if len(code_commune_clean) != 5:
+        raise ValueError("Code commune doit contenir 5 caractères !")
+    if code_commune_clean not in codes_communes:
```
Comment (XavierJp) [inline aio/aio-proxy/aio_proxy/parsers/code_commune.py:23]: Not sure what is the most efficient here.
I believe a hash would ve better than searching in an array

**Review by XavierJp — CHANGES_REQUESTED** [review]
Not sure about current validation.
How big is the code commune file
On disk ? 

I think we should have a look at performances.

Compare 100 iteration of a if code in array versus if code in  hash

File: aio/aio-proxy/aio_proxy/parsers/code_commune.py (hunk 1 above)
Comment (geoffreyaldebert) [reply aio/aio-proxy/aio_proxy/parsers/code_commune.py:23]: [a Simple regex is doing the work](https://github.com/etalab/schema-irve/blob/a74411123830283ae4ee4d4002455b54196280bb/schema.json#L205)

**Review by geoffreyaldebert — CHANGES_REQUESTED** [review]
Unless the way that code_commune is detected, nothing to say !

**Review by geoffreyaldebert — APPROVED** [review]

**Review by XavierJp — APPROVED** [review]

**Pull request merged**